server/query.py

```diff
@@ -229,6 +229,27 @@ def resultFunc():
     return resultFunc
 
 
+def csv_safe_unicode(row, ignoreErrors=True):
+    """
+    Given an array of values, make sure all strings are unicode in Python 3 and
+    str in Python 2.
```

Comment:
I don't understand this docstring... you want encoded strings in python 2 but not python 3?

Reply:
Sadly, yes.  The Python 2 `csv.writer` doesn't work with unicode, and the Python 3 writer doesn't work with binary strings.  There is an external module called `unicodecsv` which is a Python 2 replacement for the csv module which supports unicode, but this utility function is sufficient.

Reply:
:(

Let's add a note to that effect in this docstring.